Automattic/jetpack
# [Jetpack plugin] Add support for Jetpack Backup 1GB

### Changes proposed in this Pull Request:

* Add support for Jetpack Backup 1GB in the Jetpack plugin

_The new product (D81279-code) still isn't fully registered as a backup product in our own system; but this PR will ensure that we do not have to wait a full release cycle to start issuing it through partners and it will increase the likelihood of the product showing up in the future - just because we added it as early as possible._

### Jetpack product discussion



* Context: p1HpG7-feH-p2
* Part of PT: pdpAdu-kD-p2
* D81279-code

### Does this pull request change what data or activity we track or use?



No.

### Testing instructions:






* Spin up a JN site using this branch.
* Create a Jetpack Free connection.
* Go to SA and add a `Jetpack Backup 1GB` subscription to your JN site.
* Go to `/wp-admin/admin.php?page=jetpack#/my-plan`.
* Verify that the Jetpack Backup product is now showing under "Products" and says `1GB` in the description.

![Screenshot 2022-05-27 at 19 19 32](https://user-images.githubusercontent.com/3846700/170760051-27d7de6f-2e79-4b84-bd4c-b14c6892d241.png)

---

### Aside about "My Jetpack"

This PR is strictly aiming to start showing the new Jetpack Backup 1GB product in the `My Plans` area.
When Jetpack Backup 1GB is correctly registered as a backup product in our own system, "My Jetpack" should just start working because we start generating rewind data (which indirectly means that standalone plugins should also start showing the product correctly):

[Automattic\Jetpack\My_Jetpack\Products\Backup::get_state_from_wpcom](https://github.com/Automattic/jetpack/blob/master/projects/packages/my-jetpack/src/products/class-backup.php#L121-L146):
```php
private static function get_state_from_wpcom() {
	static $status = null;

	if ( $status !== null ) {
		return $status;
	}

	$site_id = Jetpack_Options::get_option( 'id' );

	$response = Client::wpcom_json_api_request_as_blog( sprintf( '/sites/%d/rewind', $site_id ) . '?force=wpcom', '2', array( 'timeout' => 2 ), null, 'wpcom' );

	if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
		return new WP_Error( 'rewind_state_fetch_failed' );
	}

	$body   = wp_remote_retrieve_body( $response );
	$status = json_decode( $body );
	return $status;
}
```
